managed-pointer/cpp-20/D0567R1.md

```diff
@@ -229,94 +391,105 @@ auto gpuExecutor = gpuContext.executor();
 auto gpuAllocator = gpuContext.allocator();
 
 /* Construct a managed_ptr ptraA allocated on the host CPU */
-std::experimental::execution::managed_ptr<float> ptrA(1024);
+execution::managed_ptr<my_data> ptrA();
 
 /* Construct a managed_ptr ptrB allocated on the GPU execution context */
-std::experimental::execution::managed_ptr<float> ptrB(1024, gpuAllocator);
+execution::managed_ptr<my_data> ptrB(gpuAllocator);
 
 /* Populate ptrA */
 populate(ptrA);
 
 /* Construct a series of compute and data operations */
 auto fut =
-  std::experimental::execution::put(gpuExecutor, ptrA)
-      .then_put(gpuExecutor, ptrB)
-          .then_async(gpuExecutor, [=](auto _ptrA, auto _ptrB) { /* ... */ }, ptrA, ptrB)
-            .then_get(gpuExecutor, ptrA, ptrB);
+  execution::put(gpuExecutor, ptrA)
+    .then_async_execute(gpuExecutor, [=]() { /* ... */ })
+      .then_get(ptrB);
```

Comment:
Yes, so the `then_get` function is essentially the `get` function with a future predicate, if it is not already accessible, it will synchronize the managed memory allocation with to host executor.